src/main/java/io/adobe/cloudmanager/impl/exception/CloudManagerExceptionDecoder.java

```diff
@@ -0,0 +1,168 @@
+package io.adobe.cloudmanager.impl.exception;
+
+/*-
+ * #%L
+ * Adobe Cloud Manager Client Library
+ * %%
+ * Copyright (C) 2020 - 2023 Adobe Inc.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Response;
+import feign.codec.ErrorDecoder;
+
+public abstract class CloudManagerExceptionDecoder implements ErrorDecoder {
+  public static final String GENERATE_BODY = "Unable to generate request body: %s.";
+
+  private static final String CONTENT_TYPE = "Content-Type";
+  private static final String APPLICATION_JSON_TYPE = "application/json";
+  private static final String DEFAULT_REASON = "Unknown";
+
+  private static final Map<Integer, String> phrases = new HashMap<>();
+
+  static {
+    phrases.put(400, "Bad Request");
```

Comment:
Maybe it might support more failure modes if we use org.apache.commons.httpclient.HttpStatus.getStatusText()?

Reply:
Don't want to add additional dependency. Wish there was another way.